stdsimd-test/src/lib.rs

```diff
@@ -293,9 +293,41 @@ pub fn assert(fnptr: usize, fnname: &str, expected: &str) {
         }
     }
 
-    let probably_only_one_instruction = function.instrs.len() < 30;
+    // Look for `call` instructions in the disassembly to detect whether
+    // inlining failed: all intrinsics are `#[inline(always)]`, so
+    // calling one intrinsic from another should not generate `call`
+    // instructions.
+    let mut inlining_failed = false;
+    for (i, instr) in function.instrs.iter().enumerate() {
+        let part = match instr.parts.get(0) {
+            Some(part) => part,
+            None => continue,
+        };
+        if !part.contains("call") {
+            continue
+        }
+
+        // On 32-bit x86 position independent code will call itself and be
+        // immediately followed by a `pop` to learn about the current address.
+        // Let's not take that into account when considering whether a function
+        // failed inlining something.
+        let followed_by_pop = function.instrs.get(i + 1)
```

Comment:
how did you figure this out?

Reply:
Oh I look at the disassembly and recognized the instruction sequence. Additionally compiling with `-C relocation-model=static` made it go away.